src/dft/backends/descriptor.cpp

```diff
@@ -28,7 +28,13 @@ namespace dft {
 
 template <precision prec, domain dom>
 void descriptor<prec, dom>::commit(sycl::queue &queue) {
-    pimpl_.reset(detail::create_commit(*this, queue));
+    if (!pimpl_ || pimpl_->get_queue() != queue) {
+        if (pimpl_) {
+            pimpl_->get_queue().wait();
+        }
+        pimpl_.reset(detail::create_commit(*this, queue));
+    }
+    pimpl_->commit(values_);
 }
```

Comment:
Maybe it's worth adding a `recommit_values` test which would change the queue between commit calls?

Reply:
I've added a tests to show that there is a wait on the old queue here https://github.com/oneapi-src/oneMKL/pull/282/commits/088fa08c85c0ebdd5baa81d22c5cb190689ac1a6
I was unable to prove that work is submitted to the new queue or not submitted to the old.
* I don't believe there is any direct way to query if a queue has not completed tasks
* I don't believe there is a way to query if an event is associated with a task.
* When an command group function object is submitted to the queue, I don't believe there are any guarantees that is will complete before slow work on other queues.
Because of all this I can't think of how to test that reliably.

Reply:
Thank you for that. I think you're right for every bullet point you mentioned and I'm sorry if my original suggestion generated complications/misunderstanding on the testing scope. Those changes look fine to me.
To be clearer, my original concern was about a use case like 
1. create a descriptor `desc`;
2. create a queue `q1` and commit `desc` to `q1`;
3. delete `q1`;
4. create a queue `q2` and commit `desc` to `q2`;
5. submit a compute task to `desc`.
I am not 100% sure this works as expected for the closed-source mkl. But that is something we can/should check on our side as that would be an issue for us to fix anyways. We may consider this thread resolved.

Reply:
Ok, that's interesting. I think this should work because objects like `sycl::queue` are kept alive by the SYCL runtime with reference counting semantics. I guess I could have a test that follows the steps you've detailed, waits on `q2` and then shows that the event returned from the task is complete. It's not perfect but its something. I'm about to finish for the day but I'll do this sometime tomorrow.

Reply:
I've added a test for that in https://github.com/oneapi-src/oneMKL/pull/282/commits/8d0a608327e5784305e216dffdf57d4060981b2d